### PR TITLE
Merging av_dev into development

### PR DESCRIPTION
**Pull request by SaintAngeLs**
Adding the couriers microservices and the orders maker

### CODE REVIEW

**Comment by SaintAngeLs** [comment]
https://github.com/SaintAngeLs/courier_app/issues/27#issue-2002782284

Basic Specification divergence is explained with the simplicity of implementaiton. If you see a whole implementation at this stage as the better way of what we should do, you may provide the update now, but anyway this changes will be required and provided in the farther steps of the development

File: SwiftParcel.Services.Couriers/src/SwiftParcel.Services.Couriers.Application/SwiftParcel.Services.Couriers.Application/Events/CourierDeleated.cs
Comment (eggwhat) [inline SwiftParcel.Services.Couriers/src/SwiftParcel.Services.Couriers.Application/SwiftParcel.Services.Couriers.Application/Events/CourierDeleated.cs:1]: There is a name typo

File: SwiftParcel.Services.Couriers/src/SwiftParcel.Services.Couriers.Application/SwiftParcel.Services.Couriers.Application/Events/CourierUpdated.cs
Comment (eggwhat) [inline SwiftParcel.Services.Couriers/src/SwiftParcel.Services.Couriers.Application/SwiftParcel.Services.Couriers.Application/Events/CourierUpdated.cs:1]: Not sure of this Guid name

**Review by an2508374 — APPROVED** [review]

File: SwiftParcel.Services.Couriers/src/SwiftParcel.Services.Couriers.Infrastructure/SwiftParcel.Services.Couriers.Infrastructure/Mongo/Queries/SearchCouriersHandler.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        private readonly IMongoRepository<CourierDocument, Guid> _repository;
+
+        public SearchCouriersHandler(IMongoRepository<CourierDocument, Guid> repository)
+            => _repository = repository;
+
+        public async Task<PagedResult<CourierDto>> HandleAsync(SearchCouriers query, CancellationToken cancellationToken)
+        {
+            PagedResult<CourierDocument> pagedResult;
+            if (query.PayloadCapacity <= 0 && query.LoadingCapacity <= 0 && query.Variants <= 0)
```
Comment (eggwhat) [inline SwiftParcel.Services.Couriers/src/SwiftParcel.Services.Couriers.Infrastructure/SwiftParcel.Services.Couriers.Infrastructure/Mongo/Queries/SearchCouriersHandler.cs:23]: Is this intended functionality?

File: SwiftParcel.Services.Couriers/src/SwiftParcel.Services.Couriers.Infrastructure/SwiftParcel.Services.Couriers.Infrastructure/Exeptions/ExceptionToMessageMapper.cs
Comment (eggwhat) [inline SwiftParcel.Services.Couriers/src/SwiftParcel.Services.Couriers.Infrastructure/SwiftParcel.Services.Couriers.Infrastructure/Exeptions/ExceptionToMessageMapper.cs:1]: Parent folder name typo

File: SwiftParcel.Services.Couriers/src/SwiftParcel.Services.Couriers.Infrastructure/SwiftParcel.Services.Couriers.Infrastructure/Mongo/Documents/Extensions.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+{
+    internal static class Extensions
+    {
+        public static Courier AsEntity(this CourierDocument document)
+            => document is null? null : new Courier(
+                document.Id,
+                // document.Brand,
+                // document.Model,
+                document.Name,
+                //document.LastName,
```
Comment (eggwhat) [inline SwiftParcel.Services.Couriers/src/SwiftParcel.Services.Couriers.Infrastructure/SwiftParcel.Services.Couriers.Infrastructure/Mongo/Documents/Extensions.cs:18]: Do we use 'Name' prop for Courier's fullname?

File: SwiftParcel.Services.OrdersCreator/src/SwiftParcel.Services.OrdersCreator/Services/Clients/CouriersServiceClient.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        {
+            _httpClient = httpClient;
+            _url = options.Services["couriers"];
+        }
+
+        public async Task<CourierDto> GetBestAsync()
+        {
+            // the most potential resource of the issues to resolve in the api connection
+            var vehicles = await _httpClient.GetAsync<PagedResult<CourierDto>>($"{_url}/courier");
+            var bestVehicle = vehicles?.Items?.FirstOrDefault(); // typical AI in a startup
```
Comment (eggwhat) [inline SwiftParcel.Services.OrdersCreator/src/SwiftParcel.Services.OrdersCreator/Services/Clients/CouriersServiceClient.cs:26]: Wrong name I guess?

File: SwiftParcel.Services.OrdersCreator/src/SwiftParcel.Services.OrdersCreator/Services/Clients/CouriersServiceClient.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        public CouriersServiceClient(IHttpClient httpClient, HttpClientOptions options)
+        {
+            _httpClient = httpClient;
+            _url = options.Services["couriers"];
+        }
+
+        public async Task<CourierDto> GetBestAsync()
+        {
+            // the most potential resource of the issues to resolve in the api connection
+            var vehicles = await _httpClient.GetAsync<PagedResult<CourierDto>>($"{_url}/courier");
```
Comment (eggwhat) [inline SwiftParcel.Services.OrdersCreator/src/SwiftParcel.Services.OrdersCreator/Services/Clients/CouriersServiceClient.cs:25]: Not sure what this name is doing here in this context.

**Review by eggwhat — APPROVED** [review]
Except for the things that I have already meantioned, everything seems completely fine. Good job

**Comment by eggwhat** [comment]
I found out that SwiftParcel.Services.Availability.Api is missing connection string for mongo in appsettings.json

**Comment by eggwhat** [comment]
> I found out that SwiftParcel.Services.Availability.Api is missing connection string for mongo in appsettings.json

I took care of that

**Review by eggwhat — APPROVED** [review]
Seems fine

**Comment by SaintAngeLs** [comment]
> > I found out that SwiftParcel.Services.Availability.Api is missing connection string for mongo in appsettings.json
> 
> I took care of that

Thank you

**Pull request merged**